src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -806,13 +806,13 @@ amountFromJson (SField const& name, Json::Value const& v)
     Json::Value currency;
     Json::Value issuer;
 
-    if (v.isObject ())
+    if (v.isObjectOrNull ())
```

Comment:
Hmm... so if `v` isn't an `object` it will be made into one inside this loop.

Reply:
I believe if `v` is null, `currency` gets set to an empty string, causing `native` to be true.   This will subsequently throw a `runtime_error` on line 850.